floris/simulation/wake_deflection/gauss.py

```diff
@@ -267,82 +362,46 @@ def wake_added_yaw(
     """
 
     # turbine parameters
-    D = rotor_diameter              # scalar
-    HH = hub_height                 # scalar
-    Ct = ct_i                       # (wd, ws, 1, 1, 1) for the current turbine
-    TSR = tip_speed_ratio           # scalar
-    aI = axial_induction_i          # (wd, ws, 1, 1, 1) for the current turbine
-    avg_v = np.mean(v_i, axis=(3,4))  # (wd, ws, 1, grid, grid)
+    D = rotor_diameter  # scalar
+    HH = hub_height  # scalar
+    Ct = ct_i  # (wd, ws, 1, 1, 1) for the current turbine
+    TSR = tip_speed_ratio  # scalar
+    aI = axial_induction_i  # (wd, ws, 1, 1, 1) for the current turbine
+    avg_v = np.mean(v_i, axis=(3, 4))  # (wd, ws, 1, grid, grid)
 
     # flow parameters
-    Uinf = np.mean(u_initial, axis=(2,3,4))
-    Uinf = Uinf[:,:,None,None,None]
+    Uinf = np.mean(u_initial, axis=(2, 3, 4))
+    Uinf = Uinf[:, :, None, None, None]
 
     # TODO: Allow user input for eps gain
     eps_gain = 0.2
     eps = eps_gain * D  # Use set value
 
-    vel_top = ((HH + D / 2) / HH) ** 0.12 * np.ones((1, 1, 1, 1, 1))
-    Gamma_top = gamma(
-        D,
-        vel_top,
-        Uinf,
-        Ct,
-        scale,
+    Gamma_top, Gamma_bottom, Gamma_wake_rotation = _calculate_gamma(
+        HH, D, Uinf, Ct, scale, u_i, aI, TSR
     )
 
-    vel_bottom = ((HH - D / 2) / HH) ** 0.12 * np.ones((1, 1, 1, 1, 1))
-    Gamma_bottom = -1 * gamma(
-        D,
-        vel_bottom,
-        Uinf,
-        Ct,
-        scale,
-    )
-
-    turbine_average_velocity = np.cbrt(np.mean(u_i ** 3, axis=(3,4)))
-    turbine_average_velocity = turbine_average_velocity[:,:,:,None,None]
-    Gamma_wake_rotation = 0.25 * 2 * np.pi * D * (aI - aI ** 2) * turbine_average_velocity / TSR
-
-    ### compute the spanwise and vertical velocities induced by yaw
+    # compute the spanwise and vertical velocities induced by yaw
 
     # decay = eps ** 2 / (4 * nu * delta_x / Uinf + eps ** 2)   # This is the decay downstream
     yLocs = delta_y + BaseModel.NUM_EPS
 
     # top vortex
-    # NOTE: this is the top of the grid, not the top of the rotor
-    zT = z_i - (HH + D / 2) + BaseModel.NUM_EPS  # distance from the top of the grid
-    rT = yLocs ** 2 + zT ** 2  # TODO: This is - in the paper
-    # This looks like spanwise decay;
-    # it defines the vortex profile in the spanwise directions
-    core_shape = 1 - np.exp(-rT / (eps ** 2))
-    v_top = (Gamma_top * zT) / (2 * np.pi * rT) * core_shape
-    v_top = np.mean( v_top, axis=(3,4) )
-    # w_top = (-1 * Gamma_top * yLocs) / (2 * np.pi * rT) * core_shape * decay
+    v_top, _ = _calculate_vortex(z_i, HH, D, yLocs, eps, Gamma_top, which="top")
```

Comment:
@RHammond2 I see and appreciate your intent here. In the original code, there's a lot of repeat and it can clearly be consolidated into a function. However, I find that readability is reduced, in this particular case. In comparing to the paper, you can follow along with what the original code is calculating, but as a reader you have to parse the new function and map it back to the variables in the calling code. Additionally, it was really helpful to print and plot different terms from each vortex calculation during development, so it would be good to preserve that functionality for future debugging. Instead of consolidation to a function, I propose to incorporate numexpr into all of the terms. I'm happy to make those changes, but do you agree with that suggestion?

Reply:
Thanks for the feedback, @rafmudaf! I'm totally fine with switching this back given that it helps align the methods with the paper that most users would be familiar with. If you're feeling particularly motivated, then you're more than welcome to put this back in place, otherwise, I'm ok with switching it back myself.